lambda/translcore.ml

```diff
@@ -1119,7 +1115,7 @@ and transl_function0
       match cases with
       | [] ->
         (* With Camlp4, a pattern matching might be empty *)
-        Alloc_heap, Pgenval
+        alloc_heap, Pgenval
```

Comment:
(not this line, but github won't let me comment on a line you didn't change!)

I think the expression `(Tupled, tparams, return, region)` above in `transl_tupled_function` should be preceded by:
```ocaml
    let region = region || not (may_allocate_in_region body) in
```
as in `transl_function0`. This was always a valid optimisation (although we didn't previously bother optimising tupled functions in this way), but with this change it has the added benefit of ensuring that we don't create functions with `region = false` when `Config.stack_allocation` is false.

(I'm not sure whether a `region = false` function can definitely cause problems with this patch, but it seems cleaner to prevent them and this is AFAICT the only place where they can arise)